Automattic/wp-calypso
# Fix/blazepress types

Related to SELFSERVE-649

## Proposed Changes
During the Promote Post i2 implementation we left the code for fetching posts and campaigns a bit messy state. This PR puts a more appropriate typing there. In particular:
- Remove the previous implementation of getting campaigns hook `use-promote-post-campaigns-query.ts`, due to its redundancy
- Rename the new one to remove the "-new" postfix in both the file and the hook names,
- Move type definitions to `client/data/promote-post/types.ts`,
- Side fix: remove warning by turning `fill-rule` and `clip-rule` props into camel case for an SVG,
- Play with types for `getCampaignEndText` in Campaign Details,
- Put types for counting total numbers of posts and campaigns.


## Testing Instructions



- Open Calypso locally or use a live preview;
- Go through pages Tools > Advertising (it's the posts tab), then Campaigns, and, finally, Campaign Details;
- Try out infinite loading of posts and campaigns by scrolling down (you probably need a lot of items to see how it works; e.g. I have recently created about 20 test pages).

## Pre-merge Checklist
